examples/jest/__tests__/catAPI.spec.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const path = require('path')
+const { Pact, Matchers } = require('../../../dist/pact')
+const getMeCats = require('../').getMeCats
+
+describe("Cat's API", () => {
+  let url = 'http://localhost'
+
+  describe("another works", () => {
+    it('returns a successful body', async () => {
+      // add expectations
+      const EXPECTED_BODY = [
+        { cat: 2 }, { cat: 3 }
+      ]
+
+      const interaction = {
+        state: 'i have a list of cats',
+        uponReceiving: 'a request for cats with given catId',
+        withRequest: {
+          method: 'GET',
+          path: '/cats',
+          query: {
+            'catId[]': Matchers.eachLike('1'),
+          },
+          headers: {
+            'Accept': 'application/json'
+          }
+        },
+        willRespondWith: {
+          status: 200,
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: EXPECTED_BODY
+        }
+      }
+
+      await provider.addInteraction(interaction)
+
+      const response = await getMeCats({
+        url,
+        port
+      })
+
+      expect(response.headers['content-type']).toEqual('application/json')
+      expect(response.data).toEqual(EXPECTED_BODY)
+      expect(response.status).toEqual(200)
+
+      return provider.verify()
```

Comment:
It seems a bit unexpected for an `async` function to return a promise.

As above, I think the `.then()` style used in the other examples is better for separating the concepts in the test anyway.

However, if we decide to go with changing to `async/await`, I think we should change this function to also await `provider.verify()`, so as not to mix styles (and also update the other examples).

Reply:
Apologies, I wasn't very clear there. Here's a better explanation of the first paragraph: 

`provider.verify()` returns a promise, and as you mention, making a function `async` effectively turns the function into a promise, meaning you can have more readable code that deals with values instead of promise chaining. For me, this means I expect functions that are marked `async` to look like this:

```
async () => {
    ...
    return <some value that isn't a promise>
}
```

instead of:

```
async () => {
    ...
    return <some promise>
}
```

Of course, this is fixed if we `await provider.verify()` (and not needed if we support node 6 by not using `await`).

Reply:
Regardless of using async function signature or not, if you need to return a value it doesn't matter if it's a primitive, complex or a promise itself.

What I'm trying to say is that for an async function that needs to return a value it is perfectly valid to have a `return <some promise>`, or to await the value and then return it. However just running the promise or awaiting the promise without returning it will not return the value explicitly.

So even if by your reference you `await provider.verify()` you will not return the value of `provider.verify()`.